react/features/display-name/components/web/DisplayNameBadge.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+import { makeStyles } from '@material-ui/core/styles';
+import React from 'react';
+
+type Props = {
+
+    /**
+     * The name to be displayed within the badge.
+     */
+    name: string
+}
+
+const useStyles = makeStyles(theme => {
+    return {
+        badge: {
+            background: 'rgba(0, 0, 0, 0.6)',
+            borderRadius: '3px',
+            color: theme.palette.text01,
+            maxWidth: '60%',
+            overflow: 'hidden',
+            padding: '2px 16px',
+            textOverflow: 'ellipsis',
+            whiteSpace: 'nowrap'
+        }
+    };
+});
+
+/**
+ * Component that displays a name badge.
+ *
+ * @returns {ReactElement}
+ */
+const DisplayNameBadge = ({ name }: Props) => {
```

Comment:
Asking out of curiosity: Why do you use `const DisplayNameBadge = ({ name }: Props) => {` and not ` function DisplayNameBadge = ({ name }: Props) { `?

Reply:
That was just out of my ES6 habit/preference. But I changed it to our style for compatibility.